From 77c3634db989bd302d20d59cf002650da2022728 Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Wed, 10 Jul 2024 16:49:42 +0300
Subject: [PATCH] [FSW-2704] fs1: enter recovery for manifest valid failure

When both manifests validation failure happens rom should
enter in recovery. Recovery will check again if both manifests
are failure then it will enter into rom recovery.

Change-Id: I7baad2580c9b70c6110723587ee15b949e19bf6c
---

diff --git a/app/boot_rom.c b/app/boot_rom.c
index df5bca5..9926a0e 100644
--- a/app/boot_rom.c
+++ b/app/boot_rom.c
@@ -61,6 +61,12 @@
 
 	boot_rom_boot_flow_validate_manifests();
 
+	/* If both manifests are invalid enter into recovery */
+	if (!boot_rom_boot_flow_is_manifest_valid(HL_FW_SPI_MANIFEST_0) &&
+		!boot_rom_boot_flow_is_manifest_valid(HL_FW_SPI_MANIFEST_1)) {
+		execute_recovery();
+	}
+
 	boot_rom_init_boot_info();
 
 	return 0;
diff --git a/common/boot_flow.c b/common/boot_flow.c
index cb0e3f3..0670dd0 100644
--- a/common/boot_flow.c
+++ b/common/boot_flow.c
@@ -116,3 +116,11 @@
 	if (boot_rom_verify_manifest(curr_manifest_idx) == BOOT_ROM_SUCCESS)
 		secondary_manifest_valid = true;
 }
+
+bool boot_rom_boot_flow_is_manifest_valid(enum hl_fw_manifest_type man_idx)
+{
+	if (man_idx == primary_manifest_idx)
+		return primary_manifest_valid;
+	else
+		return secondary_manifest_valid;
+}
diff --git a/common/boot_flow.h b/common/boot_flow.h
index d758be6..3968321 100644
--- a/common/boot_flow.h
+++ b/common/boot_flow.h
@@ -9,5 +9,6 @@
 uint32_t boot_rom_boot_flow_prefetch_img(enum hl_fw_img_type img_type);
 void boot_rom_boot_flow_boot_ppboot(enum hl_fw_manifest_type man_idx);
 void boot_rom_boot_flow_validate_manifests(void);
+bool boot_rom_boot_flow_is_manifest_valid(enum hl_fw_manifest_type man_idx);
 
 #endif /* _BOOT_FLOW_H_ */
